src/datasources/accounts/accounts.datasource.ts

```diff
@@ -138,4 +198,39 @@ export class AccountsDatasource implements IAccountsDatasource {
       );
     }
   }
+
+  /**
+   * Returns the content from cache or executes the query and caches the result.
+   * If the specified {@link CacheDir} is empty, the query is executed and the result is cached.
+   * If the specified {@link CacheDir} is not empty, the pointed content is returned.
+   *
+   * @param cacheDir {@link CacheDir} to use for caching
+   * @param query query to execute
+   * @param ttl time to live for the cache
+   * @returns content from cache or query result
+   */
+  private async getFromCacheOrExecuteAndCache<T extends postgres.MaybeRow[]>(
+    cacheDir: CacheDir,
+    query: postgres.PendingQuery<T>,
+    ttl: number,
+  ): Promise<T> {
+    const { key, field } = cacheDir;
+    const cached = await this.cacheService.get(cacheDir);
```

Comment:
If this throws, should we instead continue with the database interaction?

Reply:
I'd say if the `cacheService.get` call throws it's OK for the whole function to throw and do not continue with the database interaction. It's the same with the rest of the datasources (if the cache retrieval fails, they won't continue with the REST requests. Wdyt? From my perspective, the cache is a hard requirement for the service so it's ok to fail.